File: app/src/main/java/com/example/tribecrypto/base/BasePresenter.java
package com.example.tribecrypto.base;

import org.jetbrains.annotations.Nullable;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BasePresenter<T> implements Presenter<T> {

    public CompositeDisposable subscriptions = new CompositeDisposable();

    private T view = null;

    public Boolean hasView() {
        return view != null;
    }

    public T getView() {
        return view;
    }

    @Override
    public void setView(@Nullable T view) {
        this.view = view;
        if(view == null) {
            subscriptions.clear();
        }
    }
}

File: app/src/main/java/com/example/tribecrypto/base/BasePresenter.kt
package com.example.tribecrypto.base

import io.reactivex.disposables.CompositeDisposable

abstract class BasePresenter<T> : Presenter<T> {

    val subscriptions: CompositeDisposable = CompositeDisposable()

    private var view: T? = null

    fun hasView(): Boolean {
        return view != null
    }

    fun getView(): T? {
        return view
    }

    override fun setView(view: T) {
        this.view = view
        if (view == null) {
            subscriptions.clear()
        }
    }
}